libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -339,7 +339,11 @@ protected Task SetPropertyValueAsync(ITurnContext turnContext, string propertyNa
         public class CachedBotState
 #pragma warning restore CA1034 // Nested types should not be visible
         {
-            internal CachedBotState(IDictionary<string, object> state = null)
+            /// <summary>
```

Comment:
If we go ahead with making this class public we should make it not nested, put it in a separate file and remove the CA1034 pragma.

Reply:
Never mind, we can't move it outside, it is already public...